components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2601,6 +2637,62 @@ private void deleteIdP(Connection conn, int tenantId, String idPName) throws SQL
         }
     }
 
+    /**
+     * Delete authentication steps involving the deleted IDP in all SPs in the given tenant.
+     *
+     * @param conn
+     * @param tenantId
+     * @param idpName
+     * @throws SQLException
+     */
+    private void deleteIdpSpAuthAssociations(Connection conn, int tenantId, String idpName) throws SQLException {
+
+        PreparedStatement removeAuthStepPreparedStatement = null;
+        PreparedStatement removeEmptyAuthStepPreparedStatement = null;
+        String removeAuthStepsSql = IdPManagementConstants.SQLQueries.DELETE_IDP_SP_AUTH_ASSOCIATIONS;
+        String removeEmptyAuthStepsSql = IdPManagementConstants.SQLQueries.REMOVE_EMPTY_SP_AUTH_STEPS;
+
+        try {
+            // Remove authentication steps in SPs with the IDP being deleted
+            removeAuthStepPreparedStatement = conn.prepareStatement(removeAuthStepsSql);
+            removeAuthStepPreparedStatement.setString(1, idpName);
+            removeAuthStepPreparedStatement.setInt(2, tenantId);
+            removeAuthStepPreparedStatement.executeUpdate();
+
+            // Clean up any empty steps left over after deletion
+            removeEmptyAuthStepPreparedStatement = conn.prepareStatement(removeEmptyAuthStepsSql);
+            removeEmptyAuthStepPreparedStatement.executeUpdate();
+        } finally {
+            IdentityDatabaseUtil.closeStatement(removeAuthStepPreparedStatement);
+            IdentityDatabaseUtil.closeStatement(removeEmptyAuthStepPreparedStatement);
+        }
+    }
+
+    /**
+     * Delete Provisioning associations of the deleted IDP with any SPs in a given tenant
+     *
+     * @param conn
+     * @param tenantId
+     * @param idpName
+     * @throws SQLException
+     */
+    private void deleteIdpSpProvisioningAssociations (Connection conn,
```

Comment:
Formatting ?

Reply:
This was intentionally formatted to make it more readable